csdc-6/adcs-simulation/cpp/src/Simulator.cpp

```diff
@@ -49,6 +53,25 @@ timestamp Simulator::update_simulation() {
     return this->simulation_time;
 }
 
+timestamp Simulator::determine_timestep() 
+{
+    //2*max error is defined as 2*0.005 degrees = 0.01 degrees
+    if (this->variableTimestep == true){
+        float calculated_timestep = (0.01*M_PI/180)/(this->system_vals.satellite.alpha_b.cwiseAbs().maxCoeff())*1000;
+        timestamp t(calculated_timestep,0);
+        this->timestep_length = t;
+        if (((float) timestep_length > maxStep/1000)) {
+            timestamp t(20,0);
+            this->timestep_length = t;
+        }else if ((float) timestep_length < minStep/1000) {
+            timestamp t(1,0);
+            this->timestep_length = t;
+        }
+    }
+
+    return timestep_length;

```

Comment:
This doesn't need a return value, you're already changing "this->timestep_length" in the function.